alluxio-ui/master/src/containers/pages/Browse/Browse.tsx

```diff
@@ -0,0 +1,255 @@
+import {IFileBlockInfo, IFileInfo} from '@alluxio/common-ui/src/constants';
+import {createDebouncedFunction, parseQuerystring} from '@alluxio/common-ui/src/utilities';
+import {IApplicationState} from '../../../store';
+import {fetchRequest} from '../../../store/browse/actions';
+import {IBrowse} from '../../../store/browse/types';
+
+import './Browse.css';
+
+interface IPropsFromState {
+  browse: IBrowse;
+  errors: string;
+  loading: boolean;
+  location: {
+    search: string;
+  };
+}
+
+interface IPropsFromDispatch {
+  fetchRequest: typeof fetchRequest;
+}
+
+interface IBrowseState {
+  end?: string;
+  limit?: string;
+  offset?: string;
+  path?: string;
+  lastFetched: {
+    end?: string;
+    limit?: string;
+    offset?: string;
+    path?: string;
+  };
+  textAreaHeight?: number;
+}
+
+type AllProps = IPropsFromState & IPropsFromDispatch;
+
+class Browse extends React.Component<AllProps, IBrowseState> {
+  private readonly textAreaResizeMs = 100;
+  private readonly debouncedUpdateTextAreaHeight = createDebouncedFunction(this.updateTextAreaHeight.bind(this), this.textAreaResizeMs, true);
+
+  constructor(props: AllProps) {
+    super(props);
+
+    const {path, offset, limit, end} = parseQuerystring(this.props.location.search);
+    this.state = {end, limit, offset, path: path || '/', lastFetched: {}};
+  }
+
+  public componentDidUpdate(prevProps: AllProps) {
+    if (this.props.location.search !== prevProps.location.search) {
+      const {path, offset, limit, end} = parseQuerystring(this.props.location.search);
+      this.setState({path, offset, limit, end});
+      this.fetchData(path, offset, limit, end);
+    }
+  }
+
+  public componentWillMount() {
+    const {path, offset, limit, end} = this.state;
+    this.fetchData(path, offset, limit, end);
+    this.updateTextAreaHeight();
+  }
+
+  public componentDidMount() {
+    window.addEventListener('resize', this.debouncedUpdateTextAreaHeight);
+  }
+
+  public componentWillUnmount() {
+    window.removeEventListener('resize', this.debouncedUpdateTextAreaHeight);
+  }
+
+  public render() {
+    const {browse, loading} = this.props;
+    let queryStringSuffix = ['offset', 'limit', 'end'].filter((key: string) => this.state[key] !== undefined)
+      .map((key: string) => `${key}=${this.state[key]}`).join('&');
+    queryStringSuffix = queryStringSuffix ? '&' + queryStringSuffix : queryStringSuffix;
+
+    if (loading) {
+      return (
+        <LoadingMessage/>
+      );
+    }
+
+    return (
+      <div className="browse-page">
+        <div className="container-fluid">
+          <div className="row">
+            <div className="col-12">
+              {!browse.currentDirectory.isDirectory && this.renderFileView(browse, queryStringSuffix)}
+              {browse.currentDirectory.isDirectory && this.renderDirectoryListing(browse.fileInfos, queryStringSuffix)}
+            </div>
+          </div>
+        </div>
+      </div>
```

Comment:
Should this kind of logic go into a `component`?

Reply:
This makes sense, will do.